argus/util.py

```diff
@@ -362,3 +367,57 @@ def get_command(command, command_type=None):
         True: WINDOWS_NANO
     }
 }
+
+
+def get_os_type(instance):
+    """Get the OS type from a `Recipe` or `TestCase` instance."""
+    # pylint: disable=protected-access
+    return instance._backend.remote_client.manager.os_type
+
+
+class skip_on_os(object):
+    """Decorator to skip a method on a specific OS."""
+
+    def __init__(self, target_os_list, reason=""):
+        """Skip the method decorated if we are using the `os_type`.
+
+        :param target_os_listos_type: A list with OS version in on which
+                                      the decorator will take effect
+        :pram
+        """
+        self._target_os_list = target_os_list
+        self._reason = reason
+
+    def __call__(self, function):
+        """Return the `function` decorated."""
+        @six.wraps(function)
+        def wrapper(*args, **kwargs):
+            try:
+                target_self = kwargs.get("self", args[0])
```

Comment:
I don't think that somebody will pass `self=self`. Taking that into consideration we can safely asume that first positional argument is a reference to the instance.